Cyborgs can now shake up compromised carbons

Justice12354 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
The Hug Module from Medical and Peacekeeper Cyborgs now removes negative stats (the same as a human) from carbons when used on Mode 1. Put in a simple way, this allows the three referred Cyborg types to wake up and remove stuns from carbons with the use of their Hug Module.

Everything else has been left untouched. This Pull Request comes related to #62369.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
As of now, the "Empowered Mode" (Mode 1) of the Hug Module only gives feedback and a false sense of utility.

While being able to help carbons that are currently laying down, in its current state, this Module gives the idea that you are able to wake people up or interrupt stuns (Feedback: "You shake (target) trying to get (target) up!").
With this change, this Cyborg tool will do what it's, intuitively, supposed to do.
Although it may sound worrying to some that Borgs will now be able to reduce stuns, knockdowns, unconsciousness, sleepiness, paralysis and immobilization, it will not affect stamina damage nor any other type of damage. The sole objective of this change is to allow Cyborgs to actually shake up carbons just like carbons do. In fact, this string was completely based on the help_shake_act, from which I copied the values as well.

Even if this change may seem like a bug fix to a bug that, to some, doesn't exist, to me, it's both a bug fix and an improvement in the consistency of this module, giving it the functionality that it deserves.
Giving an example, Medical Cyborgs constantly find themselves faced with a patient that leaves Cryo, sleeping. In this case, they cannot wake them up or help them in any way, shape or form, leaving them to an endless suffering.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
fix: Cyborgs can now shake up compromised carbons with the hugging module.
/:cl:
<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. -->

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Justice12354 committed:
Update robot_items.dm

Justice12354 committed:
Merge branch 'tgstation:master' into hugmod

phasenoisepon commented:
Previously https://github.com/tgstation/tgstation/pull/53352

Justice12354 commented:
Hey, @Rohesie. I don't understand why this PR is labelled as a Content Expansion instead of a Fix when it's fixing something that at some point broke.

Rohesie commented:
Reading the code a bit better, you're right. This is a fix, as the messages imply this should be happening.

Rohesie commented on code/game/objects/items/robot/robot_items.dm:
```diff
@@ -96,6 +96,12 @@
 		if(1)
 			if(M.health >= 0)
 				if(ishuman(M))
+					M.AdjustStun(-60)
+					M.AdjustKnockdown(-60)
+					M.AdjustUnconscious(-60)
+					M.AdjustSleeping(-100)
+					M.AdjustParalyzed(-60)
+					M.AdjustImmobilized(-60)
```
Could you turn this into a proc that both this and the  carbon `help_shake_act()` call, instead of copypasting the code?
Makes it easier to maintain it, in case someone changes the values in the future.

Justice12354 replied:
I am still learning how to code, so that might be a challenge for me, that will take some time. I am determined to get this done though.

Justice12354 committed:
status_clear now exists

status_clear cleans stun, knockdown, unconscious, sleeping, paralyzed and immobilized partially. amount multiplies by these values.

Rohesie commented on code/modules/mob/living/status_procs.dm:
```diff
@@ -379,6 +379,16 @@
 
+///////////////////////// CLEAR STATUS /////////////////////////
+
+/mob/living/proc/status_clear(amount)
+	AdjustStun(-60 * amount)
+	AdjustKnockdown(-60 * amount)
+	AdjustUnconscious(-60 * amount)
+	AdjustSleeping(-100 * amount)
+	AdjustParalyzed(-60 * amount)
+	AdjustImmobilized(-60 * amount)
```
```suggestion
/mob/living/proc/adjust_status_effects_on_shake_up()
	AdjustStun(-60)
	AdjustKnockdown(-60)
	AdjustUnconscious(-60)
	AdjustSleeping(-100)
	AdjustParalyzed(-60)
	AdjustImmobilized(-60)
```
* Use informative proc names. If it's long, it's long. The important part is for the reader to understand what is going on with the less code checking possible.
* Don't add extra unused arguments and modifiers. If anyone needs them in the future they can easily add them.

Justice12354 replied:
Got it. Thanks for the tips!

Rohesie commented on code/modules/mob/living/carbon/carbon_defense.dm:
```diff
@@ -488,12 +488,7 @@
-	AdjustSleeping(-100)
-	AdjustParalyzed(-60)
-	AdjustImmobilized(-60)
+	status_clear(1)
```
```suggestion
	adjust_status_effects_on_shake_up()
```

Rohesie commented on code/game/objects/items/robot/robot_items.dm:
```diff
@@ -96,6 +96,7 @@
 		if(1)
 			if(M.health >= 0)
 				if(ishuman(M))
+					M.status_clear(1)
```
```suggestion
					M.adjust_status_effects_on_shake_up()
```

Justice12354 committed:
Update code/modules/mob/living/status_procs.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

Justice12354 committed:
Update code/modules/mob/living/carbon/carbon_defense.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

Justice12354 committed:
Update code/game/objects/items/robot/robot_items.dm

Co-authored-by: Rohesie <rohesie@gmail.com>
